services/provider/server/server.go

```diff
@@ -553,13 +557,18 @@ func (s *OCSProviderServer) GetStorageClassClaimConfig(ctx context.Context, req
 	var storageClassName string
 	storageClassData := map[string]string{}
 
+	storageClassRequestHash := strings.TrimPrefix(storageClassRequest.Name, "storageclassrequest-")
```

Comment:
I think it is a bad idea to rely on naming conventions for required information.  

What is the exact information stored on the name that you are trying to access? and do we have a way to rebuild it directly here?

Reply:
The name generation is from [here](https://github.com/red-hat-storage/ocs-operator/blob/main/services/provider/server/storageclassrequest.go#L36-L53) and it's already being called [here](https://github.com/red-hat-storage/ocs-operator/blob/main/services/provider/server/storageclassrequest.go#L160) and being passed to us from [here](https://github.com/red-hat-storage/ocs-operator/blob/main/services/provider/server/server.go#L525). Given that the SCR name is namespaced, it's safe to assume that `ceph-client-<type>-<hash>` will also be unique within the same scope as long as `<type>` isn't duplicated. Which per our implementation it isn't.

Also, best I know I'm just implementing what was decided in the meeting on Thursday. Let me know if this was incorrect.

Reply:
Changes addressed following a discussion elsewhere.